private/model/api/load.go

```diff
@@ -225,6 +225,8 @@ func (a *API) Setup() error {
 		return err
 	}
 
+	a.addHeaderMapDocumentation()
```

Comment:
Since we know that headers are only supported by s3 (and some other handful services that I don't know of). We can have a map[string]struct of the known services with headers, for which we call addHeaderMapDocumentation(). 

This should help reduce the added checks for each service.